### PR TITLE
[T4A4][W09-B1] Ng Yong Sheng

### PR DESCRIPTION
**Pull request by JasonScourge**
Follow Interface Segregation Principle [LO-ISP]

### CODE REVIEW

**Comment by JasonScourge** [comment]
Add comments to Printable Interface Class

**Comment by JasonScourge** [comment]
Check difference between String.join and String.concat

File: src/seedu/addressbook/data/person/Person.java [hunk 1]
```diff
@@ -83,5 +83,19 @@ public int hashCode() {
     public String toString() {
         return getAsTextShowAll();
     }
+    
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    public static String getPrintableString(Printable... printables) {
+        String concatString = "";
+        for (int i = 0; i < printables.length; i++) {
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Person.java:92]: You could have used a `foreach` loop instead.

File: src/seedu/addressbook/data/person/Person.java [hunk 2]
```diff
@@ -83,5 +83,19 @@ public int hashCode() {
...
+            concatString.concat(printables[i].getPrintableString());
+            if (i < printables.length - 1) {
+                concatString.concat(", ");
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Person.java:95]: Doesn't this result in an extra `,` at the end? You could have used `StringJoiner` or `String.join` instead.

File: src/seedu/addressbook/data/person/Printable.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package seedu.addressbook.data.person;
+
+public interface Printable {
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Printable.java:3]: Missing header comment. All non trivial classes/methods should have javadoc format header comments.

**Comment by damithc** [comment]
Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**